src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -87,10 +87,12 @@ public SearchOptions Create(string resourceType, IReadOnlyList<Tuple<string, str
         {
             var searchOptions = new SearchOptions();
 
-            if (queryParameters != null && queryParameters.Any(_ => _.Item1 == KnownQueryParameterNames.GlobalEndSurrogateId && _.Item2 != null))
+            if (queryParameters != null &&
+                queryParameters.Any(x => _queryHintParameterNames.Contains(x.Item1) && x.Item2 != null))
```

Comment:
Are we correcting this condition? looks like earlier we were only looking at GlobalEndSurrogateId.
cc- @brendankowitz

Reply:
I'm going to revert those changes and simplify this PR.